### DISCUSSION TITLE
Silent Install Option

### DISCUSSION THREAD

**Discussion by dlaugh14** [coreybutler/nvm-windows#823] [General]
I know this may seem like a weird question and I feel like the packaging team should be the one's to contact someone in regards to this, but there is a packaging system in our company which makes it easy to request a package and have it installed silently in the background. Is there any way to enable a sort of silent mode via the install process so you don't get those modals in regards to "nodejs is already installed, have nvm use the already installed version", and on uninstall the warning that "nodejs will be uninstalled completely". I'm not not familiar with this process, but was hoping that there was some sort of configuration or something for this.

**Comment by coreybutler** [comment]
There is. I don't know how most companies are choosing to implement this, but I see frequently see this in bigger companies. There are two choices I can think of.

**Choice 1**: Use the standard installer and pass flags to silence the prompts by pre-answering questions. Packaging teams usually have software to automate this process. It is not a feature of the NVM4W installer.

**Choice 2**: Customize the installation with a script. There is a "no-install" option for each NVM4W release that just contains the executables and assets required to run the program. The [manual installation](https://github.com/coreybutler/nvm-windows/wiki#manual-installation) guide explains how this works. Packaging teams can use this to create their own installation scripts.
